classes/AbstractAdapter.php

```diff
@@ -127,10 +127,26 @@ public function checkToken() {
                 return false;
             }
         }
+        if ($this->hlp->getConf("mailRestriction") !== '') {
+            return $this->checkMail();
+        }
         return true;
     }
 
     /**
+     * @return bool
+     */
+    public function checkMail() {
+        $hostedDomain = $this->hlp->getConf("mailRestriction");
+        $userData = $this->getUser();
+        if (substr($userData['mail'], -strlen($hostedDomain)) === $hostedDomain) {
```

Comment:
Ah just noticed, this is not perfect, yet. I could register a domain named foocosmocode.de and it would match with cosmocode.de